### PR TITLE
Table function null default table structure

### LINKED ISSUE #46930: Default table structure for table function `null`.
**Use case**

`SELECT * FROM null()` should not require specifying the table structure. It can assume `dummy UInt8` by default.

### PR DESCRIPTION
**Pull request by ucasfl**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Use `dummy UInt8` for default structure of table function `null`. Closes https://github.com/ClickHouse/ClickHouse/issues/46930

### CODE REVIEW

File: tests/queries/0_stateless/02674_null_default_structure.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+SELECT * FROM null();
```
Comment (kssenii) [inline tests/queries/0_stateless/02674_null_default_structure.sql:1]: let's also add to test `describe null()`

File: src/TableFunctions/TableFunctionNull.h [hunk 2]
```diff
@@ -29,6 +30,7 @@ class TableFunctionNull : public ITableFunction
 
     String structure = "auto";
     ColumnsDescription structure_hint;
-};
 
+    ColumnsDescription default_structure = ColumnsDescription{NamesAndTypesList{{"dummy", std::make_shared<DataTypeUInt8>()}}};
```
Comment (kssenii) [inline src/TableFunctions/TableFunctionNull.h:34]: static const?

Comment (ucasfl) [reply src/TableFunctions/TableFunctionNull.h:34]: static const ColumnsDescription can not be initialized here, must out of line.

File: tests/queries/0_stateless/02674_null_default_structure.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
...
+DESCRIBE null();
```
Comment (kssenii) [inline tests/queries/0_stateless/02674_null_default_structure.sql:2]: Let's also add `insert into table function  null() select 1, 'str'` to make sure `structure_hint` is preferred over default structure.

File: src/TableFunctions/TableFunctionNull.h (hunk 2 above)
Comment (kssenii) [reply src/TableFunctions/TableFunctionNull.h:35]: then just const

Comment (ucasfl) [reply src/TableFunctions/TableFunctionNull.h:35]: also must be out of line.

**Review by kssenii — APPROVED** [review]

**Comment by kssenii** [comment]
> Integration tests (tsan) [5/6] — fail: 1

```
>       raise Exception("Can't wait Minio to start")
E       Exception: Can't wait Minio to start
```

**Comment by kssenii** [comment]
> Stress test (debug) — Killed by signal (in clickhouse-server.log)

not sure, it is either window view or gwp asan problem https://pastila.nl/?00119652/506fd841bc633bdfb28e377c7d736dd0

**Pull request merged**